src/pass/liveness_analysis.cc

```diff
@@ -351,10 +351,42 @@ Var LivenessAnalyzer::CreateTensorVar(const Type& type) {
   return VarCreator(this).Run(type);
 }
 
-/*! \brief Calculate the byte compact size of the given type. If the type is a tuple,
+/*!
+ * \brief Calculate the byte compact size of the given type. If the type is a tuple,
  * then the size of each tensor in the tuple will be returned. Note that size 0 means
- * a tensor with dynamic shape.
+ * a tensor with dynamic shape. If the input type contains nested tuples, the nested
+ * tuples are flattened and a flat vector of sizes is returned at the end. The order
+ * of the fields is preserved.
  */
+std::vector<int64_t> CalcBytesCompactSizes(const Type& type) {
```

Comment:
Just curious, would that be more concise if we implement this function with recursion?

Reply:
Yes, I actually moved from a recursive version to this version. Just hoping it can run a little bit faster...